### ISSUE TITLE
Fix `no-duplicate-selectors` false negatives for equivalent compound selectors

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#7869]
### What minimal example or steps are needed to reproduce the bug?

```css
.a.b {}
/* … */
.b.a {}
```

```css
#a.b {}
/* … */
.b#a {}
```

```css
div.a.b {}
/* … */
div.b.a {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "no-duplicate-selectors": true
  }
}
```

ref: https://drafts.csswg.org/selectors/#compound

**Comment by romainmenke** [comment]
This could be useful: https://github.com/csstools/postcss-plugins/blob/main/packages/selector-resolve-nested/src/compound-selector-order.ts

This sorts selectors so that we don't accidentally change the meaning after serializing.
It ensures that joining `.a` and `b` doesn't accidentally become `.ab`, but is sorted to `b.a`.

To same code could be used but we need to add an extra sort by "value" if the types are the same.

This sorting should be applied during normalization.

**Comment by ybiquitous** [comment]
@Mouvedia Thanks for the report. See also [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0BDJAjABMAvgHQDtUNt8DCB6AKkwBUALASwGdMUAbZiTZ+iAV3ZRM6GJgBOMAA4RxAFxhQANCP5y2YOfw4BPTAQhykmKhUIZUpQiCUgAZo3YwAcigC2cRDAAe7qU6QwZmYbcAgCBwBzBBBgQkxMPBBxQRhmJPhseITEkAMAWih+f0YwFAV85hgnTVl0hEw5FJhsshxQyAjGSIAxWTdymIArLgJQ2CkQxDiCBKTmOR0ndkYCOQzcgEYANiQAdiQABiTCdpwgA).

I also think this bug produces false negatives. If you have no other concerns, please make this issue ready to implent.

**Comment by Mouvedia** [comment]
I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by jeddy3** [comment]
@Mouvedia, let's keep the "status: ask to implement" label; otherwise, it generates monthly noise for anything watching the repo.
